internal/mongo/event.go

```diff
@@ -0,0 +1,55 @@
+package mongo
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+const documentKeyField = "documentKey"
+const idField = "_id"
+
+type documentKey struct {
+	ID primitive.ObjectID `bson:"_id"`
+}
+
+// check if a change event if valid
+// A valid event has at least a document key value
+func isValid(event bson.M) error {
+	_, err := documentID(event)
+	return err
+}
+
+// marshall event to an array of bytes
+func marshall(event bson.M) ([]byte, error) {
+	if err := isValid(event); err != nil {
+		return nil, fmt.Errorf("%v is not a valid change event : %w", event, err)
+	}
+	return json.Marshal(event)
+}
+
+// return the document id of the event
+func documentID(event bson.M) (string, error) {
```

Comment:
Look a bit messy but avoir to unmarshall the `bson.M` event to an `OperationLog` just to get the document ID